cartridges/app_storefront_custom/cartridge/templates/default/cart/cart.isml

```diff
@@ -0,0 +1,105 @@
+        <isif condition="${pdict.valid.error && pdict.items.length !== 0}">
+            <div class="alert alert-danger alert-dismissible valid-cart-error fade show" role="alert">
+                <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                ${pdict.valid.message}
+            </div>
+        </isif>
+    </div>
+
+    <div class="container">
+        <h1 class="page-title">${Resource.msg('title.cart', 'cart', null)}</h1>
+        <div class="row cart-header">
+            <div class="col-sm-4 hidden-xs-down">
+                <a class="continue-shopping-link" href="${URLUtils.url('Home-Show')}"
+                    title="${Resource.msg('link.continue.shopping', 'cart', null)}">
+                    ${Resource.msg('link.continue.shopping', 'cart', null)}
+                </a>
+            </div>
+            <div class="col-sm-3 text-center">
+                <h2 class="number-of-items">${Resource.msgf('label.number.items.in.cart', 'cart', null, pdict.numItems)}
+                </h2>
+            </div>
+            <div class="col-sm-5 text-right hidden-xs-down">
+                <div>
+                    <span>${Resource.msg('info.need.help', 'cart', null)}</span>
+                    <span><a class="help-phone-number"
+                            href="tel:${Resource.msg('info.phone.number', 'common', null)}">${Resource.msg('info.phone.number', 'common', null)}</a></span>
+                </div>
+            </div>
+        </div>
+        <hr class="no-margin-top">
+    </div>
+
+    <isif condition="${pdict.items.length === 0}">
+        <div class="container cart-empty">
+            <div class="row">
+                <div class="col-12 text-center">
+                    <h1>${Resource.msg('info.cart.empty.msg', 'cart', null)}</h1>
+                </div>
+            </div>
+        </div>
+    <iselse/>
+        <div class="container cart cart-page">
+            <div class="row">
+                <!---product cards--->
+                <div class="col-sm-7 col-md-8">
+                    <isloop items="${pdict.items}" var="lineItem" status="loopState">
+                        <isif condition="${lineItem.productType === 'bundle'}">
+                            <isinclude template="cart/productCard/cartBundleCard" />
+                        <iselse/>
+                            <isif condition="${lineItem.noProduct === true}">
+                                <isinclude template="cart/productCard/uncategorizedCartProductCard" />
+                            <iselse/>
+                                <isinclude template="cart/productCard/cartProductCard" />
+                            </isif>
+                        </isif>
+                        
+                        <isif condition="${pdict.numItems > 1}">
+                            <isif condition="${loopState.index === 0}">
```

Comment:
You can be more consistent, by using the same approach for checking, similarly to ${loopState.last}, you can use ${loopState.first}

Reply:
Done